AST panic: SentinelAttr

```
panic: unknown node type: 'SentinelAttr 0x346df70 <line:3571:19, col:33> 0 0'

goroutine 19 [running]:
github.com/elliotchance/c2go/ast.Parse(0xc420b01cf1, 0x31, 0x5e038d, 0x5)
	/home/lepricon/go/src/github.com/elliotchance/c2go/ast/ast.go:234 +0x386f
main.convertLinesToNodes(0xc4200a0000, 0x1995, 0x6657, 0x0, 0x0, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:96 +0x198
main.convertLinesToNodesParallel.func1.1(0xc420216000, 0xc42000c008, 0xc4200a0000, 0x1995, 0x6657, 0x0)
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:120 +0x59
created by main.convertLinesToNodesParallel.func1
	/home/lepricon/go/src/github.com/elliotchance/c2go/main.go:118 +0x12d

```

--- a/ast/ast.go
+++ b/ast/ast.go
@@ -194,6 +194,8 @@ func Parse(fullline string) Node {
 		return parseReturnStmt(line)
 	case "ReturnsTwiceAttr":
 		return parseReturnsTwiceAttr(line)
+	case "SentinelAttr":
+		return parseSentinelAttr(line)
 	case "StmtExpr":
 		return parseStmtExpr(line)
 	case "StringLiteral":

--- a/ast/position.go
+++ b/ast/position.go
@@ -334,6 +334,8 @@ func setPosition(node Node, position Position) {
 		n.Pos = position
 	case *ReturnsTwiceAttr:
 		n.Pos = position
+	case *SentinelAttr:
+		n.Pos = position
 	case *StmtExpr:
 		n.Pos = position
 	case *StringLiteral:

--- a/ast/sentinel_attr.go
+++ b/ast/sentinel_attr.go
@@ -1,0 +1,58 @@
+package ast
+
+import (
+	"strings"
+
+	"github.com/elliotchance/c2go/util"
+)
+
+type SentinelAttr struct {
+	Addr       Address
+	Pos        Position
+	A          int
+	B          int
+	ChildNodes []Node
+}
+
+func parseSentinelAttr(line string) *SentinelAttr {
+	groups := groupsFromRegex(
+		`<(?P<position>.*)>(?P<a> \d+)(?P<b> \d+)?`,
+		line,
+	)
+
+	b := 0
+	if groups["b"] != "" {
+		b = util.Atoi(strings.TrimSpace(groups["b"]))
+	}
+
+	return &SentinelAttr{
+		Addr:       ParseAddress(groups["address"]),
+		Pos:        NewPositionFromString(groups["position"]),
+		A:          util.Atoi(strings.TrimSpace(groups["a"])),
+		B:          b,
+		ChildNodes: []Node{},
+	}
+}
+
+// AddChild adds a new child node. Child nodes can then be accessed with the
+// Children attribute.
+func (n *SentinelAttr) AddChild(node Node) {
+	n.ChildNodes = append(n.ChildNodes, node)
+}
+
+// Address returns the numeric address of the node. See the documentation for
+// the Address type for more information.
+func (n *SentinelAttr) Address() Address {
+	return n.Addr
+}
+
+// Children returns the child nodes. If this node does not have any children or
+// this node does not support children it will always return an empty slice.
+func (n *SentinelAttr) Children() []Node {
+	return n.ChildNodes
+}
+
+// Position returns the position in the original source code.
+func (n *SentinelAttr) Position() Position {
+	return n.Pos
+}

--- a/ast/sentinel_attr_test.go
+++ b/ast/sentinel_attr_test.go
@@ -1,0 +1,19 @@
+package ast
+
+import (
+	"testing"
+)
+
+func TestSentinelAttr(t *testing.T) {
+	nodes := map[string]Node{
+		`0x346df70 <line:3571:19, col:33> 0 0`: &SentinelAttr{
+			Addr:       0x346df70,
+			Pos:        NewPositionFromString("line:3571:19, col:33"),
+			A:          0,
+			B:          0,
+			ChildNodes: []Node{},
+		},
+	}
+
+	runNodeTests(t, nodes)
+}

